5593 azure.resource manager.core readme (#21858)
Azure/azure-sdk-for-net: sdk/resourcemanager/Azure.ResourceManager.Core/docs/Prerequisites.md

Prerequisites
-------------

You will need the following values to authenticate to Azure

> -   **Subscription ID**

Reviewer: These are *not* prerequisites, except for a single type of authentication.  Having this as the introduction to the client is misleading.

Reviewer: For most devops scenaris, what you need is: having logged in previously using some other tool

-   **Client ID**
-   **Client Secret**
-   **Tenant ID**

These values can be obtained from the portal, here's the instructions:

### Get Subscription ID

1.  Log in to your Azure account.
2.  Select **Subscriptions** in the left sidebar.
3.  Select the subscription to be used.
4.  Select **Overview**.
5.  Copy the Subscription ID.

### Get Client ID / Client Secret / Tenant ID

For information on how to get Client ID, Client Secret, and Tenant ID, see [this
document](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal).

### Setting Environment Variables

After you obtain the values, set the following environment variables:

-   `AZURE_CLIENT_ID`
-   `AZURE_CLIENT_SECRET`
-   `AZURE_TENANT_ID`
-   `AZURE_SUBSCRIPTION_ID`

To set the environment variables on your development system:

Windows:

_(Note: Administrator access is required)_

> 

Reviewer: these are not requirements.  This is elevating the SPN + secret credential as if it is the primary credential everyone should use.  It is not.  SPN + Secret is a very specialized credential used fro some kinds of automation.  It would not be the most used credential type

Prerequisites
-------------

You will need the following values to authenticate to Azure

-   **Subscription ID**
-   **Client ID**
-   **Client Secret**
-   **Tenant ID**

These values can be obtained from the portal, here's the instructions:

### Get Subscription ID

> 1.  Login into your Azure account
> 2.  Select Subscriptions in the left sidebar
> 3.  Select whichever subscription is needed
> 4.  Click on Overview
> 5.  Copy the Subscription ID

Reviewer 2: ```suggestion
1.  Log in to your Azure account.
2.  Select **Subscriptions** in the left sidebar.
3.  Select the subscription to be used.
4.  Select **Overview**.
5.  Copy the Subscription ID.
```

### Get Client ID / Client Secret / Tenant ID

> For information on how to get Client ID, Client Secret, and Tenant ID,
> please refer to [this
> document](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal)

Reviewer 2: ```suggestion
For information on how to get Client ID, Client Secret, and Tenant ID, see [this
document](https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal).
```

### Setting Environment Variables

> After you obtained the values, you need to set the following values as
> your environment variables

Reviewer 2: ```suggestion
After you obtain the values, set the following environment variables:
```

-   `AZURE_CLIENT_ID`
-   `AZURE_CLIENT_SECRET`
-   `AZURE_TENANT_ID`
-   `AZURE_SUBSCRIPTION_ID`

> To set the following environment variables on your development system:

Reviewer 2: ```suggestion
To set the environment variables on your development system:
```

Windows:

_(Note: Administrator access is required)_

1.  Open the Control Panel
2.  Click System Security, then System
3.  Click Advanced system settings on the left
4.  Inside the System Properties window, click the Environment
    Variables button.
5.  Click on the property you would like to change, then click the Edit
    button. If the property name is not listed, then click the New
    button.

> Linux-based OS :

Reviewer 2: ```suggestion
Linux-based OS:
```
